src/protocol/oprf/oprf_share.rs

```diff
@@ -0,0 +1,148 @@
+use std::ops::Add;
+
+use generic_array::GenericArray;
+use rand::{distributions::Standard, Rng};
+use typenum::Unsigned;
+
+use super::OPRFInputRow;
+use crate::{
+    ff::{Field, Gf32Bit, Gf40Bit, Gf8Bit, Serializable},
+    helpers::{Direction, Message},
+};
+pub type OprfMK = Gf40Bit;
+pub type OprfBK = Gf8Bit;
+pub type OprfF = Gf32Bit;
+
+#[derive(Debug, Clone, Copy)]
+pub struct OPRFShare {
+    pub timestamp: OprfF,
+    pub mk: OprfMK,
+    pub is_trigger_bit: OprfF,
+    pub breakdown_key: OprfBK,
+    pub trigger_value: OprfF,
+}
+
```

Comment:
Could we implement (Weak)SharedValue for OPRFShare? Then we could operate on SharedValue during the shuffle. (WeakSharedValue is part of https://github.com/private-attribution/ipa/pull/795#top and only requires to implement additions).